routing/joint.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "routing/fseg.hpp"
+
+#include "coding/reader.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "std/cstdint.hpp"
+#include "std/limits.hpp"
+#include "std/vector.hpp"
+
+namespace routing
+{
+using JointId = uint32_t;
```

Comment:
I suggest you to move this stuff to the public section of the Join class, and rename to Id and kInvalidId.